test/export/StructureDefinition.ResourceExporter.test.ts

```diff
@@ -269,12 +270,21 @@ describe('ResourceExporter', () => {
         /FHIR prohibits logical models and resources from constraining parent elements. Skipping.*at path 'language'.*File: ConstrainParent\.fsh.*Line:\D*/s
       );
     });
+  });
 
-    expect(exported.name).toBe('MyTestResource');
-    expect(exported.id).toBe('MyResource');
-    expect(exported.type).toBe('MyResource');
-    expect(exported.baseDefinition).toBe('http://hl7.org/fhir/StructureDefinition/DomainResource');
-    expect(exported.elements).toHaveLength(12); // 9 AlternateIdentification elements + 3 added elements
+  it('should log an error when adding an element with the same path as an inherited element', () => {
+    const resource = new Resource('MyResource');
+    const addElementRule = new AddElementRule('extension');
+    addElementRule.min = 0;
+    addElementRule.max = '1';
+    resource.rules.push(addElementRule);
+    doc.resources.set(resource.name, resource);
+    exporter.export();
```

Comment:
I think it would be good to also include a check within this test that a new `extension` element is not added. I think you can do this by grabbing the exported resource (there is an example of that in the test above this one) and checking the elements array. Without the error you added in, there would be two elements with the id `MyResource.extension`. You could check the total length of the `elements` array and check that there is only one element with that path. There may be other/additional ways to check that, but those are the two ideas I thought of.